gluten-core/src/main/java/io/glutenproject/substrait/rel/RelBuilder.java

```diff
@@ -108,6 +108,20 @@ public static RelNode makeAggregateRel(RelNode input,
     return new AggregateRelNode(input, groupings, aggregateFunctionNodes, filters, extensionNode);
   }
 
+  public static RelNode makeReadRelWithLocalFile(ArrayList<TypeNode> types,
+                                                 ArrayList<String> names,
+                                                 ExpressionNode filter,
+                                                 SubstraitContext context,
+                                                 Long operatorId) {
```

Comment:
For the replacement from `makeReadRel` to `makeReadRelWithLocalFile`, can we ensure the logics being the same as below for Velox backend? And if below function is replaced, can we remove it?

https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/java/io/glutenproject/substrait/rel/RelBuilder.java#L151-L175

The name is also some kind of misleading, because this function only wraps an iterator into LocalFiles.

Reply:
I check all the places which are replaced in the shared code,  all of them use the plan.output directly. It will be OK for `velox`

Reply:
What about below code? It is for metrics updating.

```
    if (operatorId >= 0) {
      // If the operator id is negative, will not register the rel to operator.
      // Currently, only for the special handling in join.
      context.registerRelToOperator(operatorId);
    }
```

Why not keep only one of the two functions in Gluten? A little confusing to me.

Reply:
where is this

Reply:
It is the previous call in ` val readRel = RelBuilder.makeReadRel(attrList, context, operatorId)`. I guess all the usages in Gluten need to be changed due to API change.